libs/accounts/src/lib/accounts-manager.tsx

```diff
@@ -21,26 +24,29 @@ export const AccountsManager = ({ partyId }: AccountsManagerProps) => {
   const gridRef = useRef<AgGridReact | null>(null);
   const variables = useMemo(() => ({ partyId }), [partyId]);
   const update = useCallback(
-    ({ delta }: { delta: AccountFieldsFragment }) => {
+    ({ delta: deltas }: { delta: AccountEventsSubscription['accounts'] }) => {
       const update: AccountFieldsFragment[] = [];
       const add: AccountFieldsFragment[] = [];
       if (!gridRef.current?.api) {
         return false;
       }
-      const rowNode = gridRef.current.api.getRowNode(getId(delta));
-      if (rowNode) {
-        const updatedData = produce<AccountFieldsFragment>(
-          rowNode.data,
-          (draft: AccountFieldsFragment) => {
-            merge(draft, delta);
+      const api = gridRef.current.api;
+      deltas.forEach((delta) => {
+        const rowNode = api.getRowNode(getId(delta));
+        if (rowNode) {
+          const updatedData = produce<AccountFieldsFragment>(
+            rowNode.data,
+            (draft: AccountFieldsFragment) => {
+              merge(draft, delta);
+            }
+          );
+          if (updatedData !== rowNode.data) {
+            update.push(updatedData);
           }
-        );
-        if (updatedData !== rowNode.data) {
-          update.push(updatedData);
+        } else {
+          // #TODO handle new account (or leave it to data provider to handle it)
```

Comment:
TODO?

Reply:
It's hanging there for a while. I created separate user story for this.
https://github.com/vegaprotocol/frontend-monorepo/issues/1391